src/main/java/org/eluder/coveralls/maven/plugin/parser/JaCoCoParser.java

```diff
@@ -58,10 +58,22 @@ protected void onEvent(final XMLStreamReader xml, final SourceCallback callback)
         
         if (isStartElement(xml, "line") && this.source != null) {
             int ci = Integer.parseInt(xml.getAttributeValue(null, "ci"));
-            this.source.addCoverage(
-                    Integer.parseInt(xml.getAttributeValue(null, "nr")),
-                    (ci == 0 ? 0 : 1) // jacoco does not count hits
-            );
+            int cb = Integer.parseInt(xml.getAttributeValue(null, "cb"));
+            int mb = Integer.parseInt(xml.getAttributeValue(null, "mb"));
+            int nr = Integer.parseInt(xml.getAttributeValue(null, "nr"));
+
+            // jacoco does not count hits. this is why hits is always 0 or 1
+            this.source.addCoverage(nr, (ci == 0 ? 0 : 1));
+
+            // add branches. unfortunately, there is NO block number and
+            // branch number will NOT be unique between coverage changes.
+            int branchId = 0;
```

Comment:
Probably is better to use an unique ID per source file or session instead of an unique ID per condition.